src/Microsoft.AspNet.OData.Shared/Builder/ODataInstanceAnnotationContainer.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.AspNet.OData.Builder
+{
+    /// <summary>
+    /// Default implementation for IODataInstanceAnnotationContainer
+    /// </summary>
+    public class ODataInstanceAnnotationContainer : IODataInstanceAnnotationContainer
+    {
+        private IDictionary<string, IDictionary<string, object>> instanceAnnotations;
+
+
+        /// <summary>
+        /// Initializes a new instance of the class
+        /// </summary>
+        public ODataInstanceAnnotationContainer()
+        {
+            instanceAnnotations = new Dictionary<string, IDictionary<string, object>>();
+        }
+
+        /// <summary>
+        /// Method to Add an Instance Annotation to the CLR type
+        /// </summary>
+        /// <param name="annotationName">Name of Annotation</param>
+        /// <param name="value">Value of Annotation</param>
+        public void AddResourceAnnotation(string annotationName, object value)
```

Comment:
>void [](start = 15, length = 4)

virtual

Reply:
DId not change this syntax?  it's better to make these method as vritual?

---
In reply to: [490484467](https://github.com/OData/WebApi/pull/2219#discussion_r490484467) [](ancestors = 490484467)

Reply:
For now we'll keep them void and seal the class until we can understand how the developer could derive from the class without messing up internal state.

---
In reply to: [490585741](https://github.com/OData/WebApi/pull/2219#discussion_r490585741) [](ancestors = 490585741,490484467)